app/cdap/components/WranglerGrid/SelectColumnPanel/DataTable/types.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import {
+  IHeaderNamesList,
+  IDataQualityItem,
+} from 'components/WranglerGrid/SelectColumnPanel/types';
+
+export interface IColumnTableProps {
+  columns: IHeaderNamesList[];
+  transformationDataType: string[];
+  onSingleSelection: (value: IHeaderNamesList) => void;
+  selectedColumns: IHeaderNamesList[];
+  dataQualityValue: IDataQualityItem[];
+  isSingleSelection: boolean;
+  handleDisableCheckbox: () => boolean;
+  onMultipleSelection: (
+    event: React.ChangeEvent<HTMLInputElement>,
+    value: IHeaderNamesList
+  ) => void;
+  totalColumnCount: number;
+  setSelectedColumns: React.Dispatch<React.SetStateAction<IHeaderNamesList[]>>;
+  transformationName: string;
+}
+
+export interface ITableRowProps {
```

Comment:
plz make sure the interfaces are not redundant i.e. we have written them only once and we have used extends wherever we can, 
I suppose extends should be used between relative interface w.r.t. an entity they are referring
plz do not mix up different entities interfaces unless they can be generalized properly

plz understand carefully and implement

Reply:
Done